Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/DebugLogViewModel.kt

```diff
@@ -0,0 +1,115 @@
+package de.rki.coronawarnapp.bugreporting.debuglog.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.bugreporting.debuglog.DebugLogger
+import de.rki.coronawarnapp.nearby.ENFClient
+import de.rki.coronawarnapp.util.CWADebug
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.compression.Zipper
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.sharing.FileSharing
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flow
+import org.joda.time.format.DateTimeFormat
+import timber.log.Timber
+import java.io.File
+
+class DebugLogViewModel @AssistedInject constructor(
+    private val debugLogger: DebugLogger,
+    dispatcherProvider: DispatcherProvider,
+    private val timeStamper: TimeStamper,
+    private val fileSharing: FileSharing,
+    private val enfClient: ENFClient
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+    private val ticker = flow {
```

Comment:
there is a nice tickerchannel in kotlin. has some little benefits.. (highly optional but :cherries:)
have a look here:
https://kotlinlang.org/docs/reference/coroutines/channels.html#ticker-channels

Reply:
I tried that but it seems this is deprecated/experimental?
I got warnings and would have had to annotate using this.
I'll look into it again.